### PR TITLE
[FIX] mail: remove auto isCausal on inverses of required fields

### PR DESCRIPTION
**Pull request by oomsveta**
Previously, relational required fields had their auto-generated inverses defined with `isCausal: true`, which seemed logical: if a record is deleted, all the records that depend on it must be deleted too.

In practice, it turned out to be a bad idea since there are some cases where you don't want the dependent records to be deleted. For example, there is no point in deleting the current record when a required field is replaced by another one.

Furthermore, this actually led to the introduction of new bugs, as the one described in task-2730667.

Fixes bug described in task-2730667.
Reverts some of these changes: https://github.com/odoo/odoo/pull/81679

### CODE REVIEW

**Review by seb-odoo — CHANGES_REQUESTED** [review]
missing adding is causal back on:
- attachmentViewers of attachment
- attachmentViewers of attachmentList
- messageSeenIndicators of message
- partnerSeenInfos of partner

and need to remove the bad is causal:
- thread on threadCache
- chatter on threadViewer
- chatWindow on threadViewer
- discuss on threadViewer
- notificationListViewOwner on the various inverses that have it

And maybe more, need to double check the original PR and revert it + check the fields that have been introduced since

File: addons/mail/static/src/model/model_manager.js [hunk 1]
```diff
@@ -1069,9 +1066,10 @@ export class ModelManager {
             relFunc(Model.name, { inverse: field.fieldName }),
             {
                 fieldName: `_inverse_${Model}/${field.fieldName}`,
-                // Note that an identifying field is not necessarily defined as
-                // `required` (example: identifyingFields with an OR). 
-                isCausal: field.required || Model.__identifyingFieldsFlattened.has(field.fieldName),
+                // Add isCausal to inverses that are identifying fields. This is
+                // needed to support models that aren't directly defined in
+                // JavaScript.
```
[the remark this answers is not in the document]
Comment (oomsveta) [reply addons/mail/static/src/model/model_manager.js:1071]: I feel like a comment is still relevant here. If I wasn't the one who wrote that code, I probably wouldn't have understood why this line is necessary. But apart from that I agree with you, it's too specific and it would be better to at least reword it to be more general.

**Comment by oomsveta** [comment]
> missing adding is causal back on:
> 
>     * attachmentViewers of attachment
> 
>     * attachmentViewers of attachmentList
> 
>     * messageSeenIndicators of message
> 
>     * partnerSeenInfos of partner

This was intentional, `attachmentViewers` on `attachment` seems to be useless since the beginning, and the others are inverses of identiyfing fields, meaning that they still don't need to be defined.

**Comment by seb-odoo** [comment]
They are necessary, it makes no sense to have an attachment view of an attachment that doesn't exist.

**Comment by seb-odoo** [comment]
But they too are inverses of identifying, so ok to not add them actually.

**Comment by oomsveta** [comment]
> They are necessary, it makes no sense to have an attachment view of an attachment that doesn't exist.

`attachmentViewers` on `attachment` would be the very first many2many field in our codebase to have isCausal set to true then. I'm not sure we're talking about the same thing.

**Comment by seb-odoo** [comment]
Indeed forget it, it was an unrelated change done in the other PR that lead to confusion (removing obsolete field in a PR that did something else).

File: addons/mail/static/src/models/attachment/attachment.js [hunk 2]
```diff
@@ -279,6 +279,20 @@ registerModel({
...
+            inverse: 'attachment',
+            isCausal: true,
+        }),
+        /**
+         * States the attachment images that are displaying this attachment.
+         */
+        attachmentImages: one2many('AttachmentImage', {
+            inverse: 'attachment',
+            isCausal: true,
+        }),
```
Comment (seb-odoo) [inline addons/mail/static/src/models/attachment/attachment.js:295]: these are inverse of identifying too, so also don't need to add them back?

**Comment by oomsveta** [comment]
Also, I couldn't find any occurrence of `notificationListViewOwner` that wasn't used as an identifying field, so I guess `isCausal` is justified on all the inverses. If you have no further remarks, I'm done with all your comments 🙂

File: addons/mail/static/src/models/composer_view/composer_view.js [hunk 3]
```diff
@@ -953,6 +953,7 @@ registerModel({
          */
         composer: many2one('Composer', {
             compute: '_computeComposer',
+            inverse: 'composerViews',
```
Comment (seb-odoo) [inline addons/mail/static/src/models/composer_view/composer_view.js:956]: Not for this PR, but maybe to investigate, it is a bit strange that composer is not an identifying field in composer view, if we follow the usual logic of view components it is not natural. But I guess it is a side issue of composition instead of trait here?

I'm actually wondering in general if inverse fields of `isCausal` shouldn't always be identifying fields on the inverse model. I think that would actually make sense? So pretty much remove `isCausal`, and only make the "inverse of is causal" on all identifying fields?

Would be interesting to list the `isCausal` we have and see which are not actually inverse of identifying fields, and if those could become identifying fields.

**Review by seb-odoo — APPROVED** [review]
@robodoo r+
